huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go

```diff
@@ -473,7 +578,7 @@ func resourceDmsRabbitmqInstanceUpdate(ctx context.Context, d *schema.ResourceDa
 		}
 	}
 
-	if d.HasChange("product_id") {
+	if d.HasChanges("product_id", "flavor_id") {
 		err = resizeRabbitMQInstance(ctx, d, meta, engineRabbitMQ)
```

Comment:
split this function to `resizeRabbitMQInstanceByFlavorId` and `resizeRabbitMQInstanceByProductId`?

Reply:
they are both modify flavor, and the param of broker_num and storage_space will also support modification, so it is better to put them together